app/assets/javascripts/components/image-cropper.js

```diff
@@ -0,0 +1,78 @@
+//= require cropperjs/dist/cropper.js
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function ImageCropper ($imageCropper) {
+    this.$imageCropper = $imageCropper
+    this.$image = this.$imageCropper.querySelector('.app-c-image-cropper__image')
+  }
+
+  ImageCropper.prototype.init = function () {
+    // This only runs if the image isn't cached
+    this.$image.addEventListener('load', function () {
+      this.initCropper()
+    }.bind(this))
+
+    // This should only run if the image is cached
+    if (this.$image.complete) {
+      this.initCropper()
+    }
+
+    this.setupFormListener()
+  }
+
+  ImageCropper.prototype.initCropper = function () {
+    if (!this.$image.complete) {
+      return
+    }
+
+    var width = this.$image.clientWidth
+    var naturalWidth = this.$image.naturalWidth
+    var scaledRatio = width / naturalWidth
+
+    // Adjust the crop box limits to the scaled image
+    var minCropBoxWidth = Math.round(960 * scaledRatio)
+    var minCropBoxHeight = Math.round(640 * scaledRatio)
+
+    if (this.$image) {
+      this.cropper = new window.Cropper(this.$image, { // eslint-disable-line
+        viewMode: 2,
+        aspectRatio: 3 / 2,
+        autoCrop: true,
+        autoCropArea: 1,
+        guides: false,
+        zoomable: false,
+        highlight: false,
+        minCropBoxWidth: minCropBoxWidth,
+        minCropBoxHeight: minCropBoxHeight,
+        rotatable: false,
+        scalable: false
+      })
+    }
+  }
+
+  ImageCropper.prototype.setupFormListener = function () {
+    var input = this.$imageCropper.querySelector('.js-cropped-image-input')
+    input.form.addEventListener('submit', function (event) {
+      event.preventDefault()
+      this.cropper.getCroppedCanvas({
+        width: 960
+      }).toBlob(function (blob) {
+        var file = new File(
+          [blob],
+          this.$imageCropper.dataset.filename,
+          {
+            type: 'image/jpeg',
+            lastModified: new Date().getTime()
+          })
+        var container = new DataTransfer()
+        container.items.add(file)
+        input.files = container.files
+        input.form.submit()
+      }.bind(this), 'image/jpeg', 0.8)
```

Comment:
What about images that were uploaded as PNGs or GIFs?

It would be nice to pass the original mime type in to [HTMLCanvasElement.toBlob()](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob) so that we don't inadvertently end up turning crisp PNGs into blurry JPEGs.

Reply:
Also, minor side note but hopefully not a blocker: it looks like Safari [doesn't support the 'quality' parameter](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob#browser_compatibility). So presumably (?) JPEGs are just exported at 100% quality 🤷🏻

Post-processing [happens server-side anyway](https://github.com/alphagov/whitehall/blob/6274099c39bab0d84761a2b549c703f56814eab3/app/uploaders/image_uploader.rb#L14-L16) so this shouldn't result in uncompressed JPEGs being published to GOV.UK in practice.

In fact, might even be worth using 100% quality all the time anyway, to avoid double-compressing the images. 🤷🏻 But I don't have a strong opinion on that.

Reply:
Have added code to maintain the type and removed the compression parameter 👍